src/components/AdminSettings.vue

```diff
@@ -82,36 +82,95 @@
 						@update:value="onUnsignedIntFieldChanged(state.max_image_generation_idle_time, 'max_image_generation_idle_time')" />
 				</div>
 			</div>
+			<div class="chat-with-ai">
+				<h2>
+					{{ t('assistant', 'Chat with AI') }}
+				</h2>
+				<div class="line">
+					<label for="chat_user_instructions">
+						{{ t('assistant', 'Chat User Instructions for Chat Completions') }}
+					</label>
+				</div>
+				<NcNoteCard type="info">
+					<p>{{ t('assistant', 'It is passed on to the LLM for it to better understand the context.') }}</p>
+					<p>{{ t('assistant', '"{user}" is a placeholder for the user\'s display name.') }}</p>
+				</NcNoteCard>
+				<NcRichContenteditable id="chat_user_instructions"
+					class="text-field"
+					:value.sync="state.chat_user_instructions"
+					:auto-complete="() => {}"
+					:link-auto-complete="false"
+					:placeholder="t('assistant', 'Chat User Instructions')"
+					:aria-label="t('assistant', 'Chat User Instructions')"
+					dir="auto"
+					@update:value="delayedValueUpdate(state.chat_user_instructions, 'chat_user_instructions')"
```

Comment:
It is already debounced though.

```js
		delayedValueUpdate(newValue, key) {
			this.optionsToSave[key] = newValue

			if (this.textFieldSaveTimer) {
				clearTimeout(this.textFieldSaveTimer)
			}
			this.textFieldSaveTimer = setTimeout(() => {
				this.saveOptions(this.optionsToSave)
			}, 2000)
		},
```

Reply:
I think julien means you can save some code by using the 'debounce' package

Reply:
makes it easier to read, too

Reply:
I guess it would increase the amount of code if we use a package and doesn't really have much impact on functionality here imo. We can extract it into a util like this, still keeping it local: https://github.com/nextcloud/integration_mattermost/blob/main/src/utils.js#L8-L18

Reply:
Yeah fine if you prefer using your own debounce implementation. Just put it in a util file.